ivy/functional/ivy/device.py

```diff
@@ -267,6 +283,66 @@ def print_all_ivy_arrays_on_dev(
         [print(arr) for arr in arrs]
 
 
+@handle_exceptions
+def set_soft_device_mode(mode: bool) -> None:
+    """
+    Set the mode of whether to move input arrays to `ivy.default_device()` before
+    performing an operation.
+    Parameter
+    ---------
+    mode
+        boolean whether to move input arrays
+    Examples
+    --------
+    >>> ivy.set_soft_device_mode(False)
+    >>> ivy.get_soft_device_mode()
+    False
+    >>> ivy.set_soft_device_mode(True)
+    >>> ivy.get_soft_device_mode()
+    True
+    """
+    global soft_device_mode_stack
+    ivy.utils.assertions.check_isinstance(mode, bool)
+    soft_device_mode_stack.append(mode)
+
+
+@handle_exceptions
+def unset_soft_device_mode() -> None:
+    """
+    Reset the mode of moving input arrays to `ivy.default_device()` before performing an
+    operation.
+    Examples
+    --------
+    >>> ivy.set_soft_device_mode(False)
+    >>> ivy.get_soft_device_mode()
+    False
+    >>> ivy.unset_soft_device_mode()
+    >>> ivy.get_soft_device_mode()
+    True
+    """
+    global soft_device_mode_stack
+    if soft_device_mode_stack:
+        soft_device_mode_stack.pop(-1)
+
+
+@handle_exceptions
+def get_soft_device_mode() -> bool:
```

Comment:
Given that we've now replaced all getter methods for modes by [properties](https://github.com/unifyai/ivy/commit/f23a27b5128b713fc3e317de32f591fe05724928), we'd need to incorporate that in this mode too